### ISSUE TITLE
Check failed: stats.is_op_tensor_core_eligible == kernel_report.is_op_tensor_core_eligible() (1 vs. 0)

### ISSUE THREAD

**Issue by j-irion** [tensorflow/tensorboard#6976]
### Diagnostics

<details>
<summary>Diagnostics output</summary>

``````
--- check: autoidentify
INFO: diagnose_tensorboard.py version c6ca9f1d004e2a1bc7c160abc43be229b82cad7e

--- check: general
INFO: sys.version_info: sys.version_info(major=3, minor=10, micro=12, releaselevel='final', serial=0)
INFO: os.name: posix
INFO: os.uname(): posix.uname_result(sysname='Linux', nodename=[REDACTED_HOSTNAME], release='5.15.0-131-generic', version='#141-Ubuntu SMP Fri Jan 10 21:18:28 UTC 2025', machine='x86_64')
INFO: sys.getwindowsversion(): N/A

--- check: package_management
INFO: has conda-meta: False
INFO: $VIRTUAL_ENV: '[REDACTED_PATH]/.venv'

--- check: installed_packages
INFO: installed: tensorboard==2.18.0
INFO: installed: tensorflow==2.18.0
WARNING: no installation among: ['tensorflow-estimator', 'tensorflow-estimator-2.0-preview', 'tf-estimator-nightly']
INFO: installed: tensorboard-data-server==0.7.2

--- check: tensorboard_python_version
INFO: tensorboard.version.VERSION: '2.18.0'

--- check: tensorflow_python_version
2025-02-19 10:51:09.119228: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2025-02-19 10:51:09.131751: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:477] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
E0000 00:00:1739962269.146209 4174031 cuda_dnn.cc:8310] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
E0000 00:00:1739962269.150586 4174031 cuda_blas.cc:1418] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2025-02-19 10:51:09.166389: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI AVX512_BF16 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
INFO: tensorflow.__version__: '2.18.0'
INFO: tensorflow.__git_version__: 'v2.18.0-rc2-4-g6550e4bd802'

--- check: tensorboard_data_server_version
INFO: data server binary: '[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard_data_server/bin/server'
INFO: data server binary version: b'rustboard 0.7.2'

--- check: tensorboard_binary_path
INFO: which tensorboard: b'[REDACTED_PATH]/.venv/bin/tensorboard\n'

--- check: addrinfos
socket.has_ipv6 = True
socket.AF_UNSPEC = <AddressFamily.AF_UNSPEC: 0>
socket.SOCK_STREAM = <SocketKind.SOCK_STREAM: 1>
socket.AI_ADDRCONFIG = <AddressInfo.AI_ADDRCONFIG: 32>
socket.AI_PASSIVE = <AddressInfo.AI_PASSIVE: 1>
Loopback flags: <AddressInfo.AI_ADDRCONFIG: 32>
Loopback infos: [(<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::1', 0, 0, 0)), (<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 0))]
Wildcard flags: <AddressInfo.AI_PASSIVE: 1>
Wildcard infos: [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('0.0.0.0', 0)), (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::', 0, 0, 0))]

--- check: readable_fqdn
INFO: socket.getfqdn(): [REDACTED_HOSTNAME]

--- check: stat_tensorboardinfo
INFO: directory: /tmp/.tensorboard-info
INFO: os.stat(...): os.stat_result(st_mode=16895, st_ino=198705172, st_dev=66306, st_nlink=2, st_uid=1000, st_gid=1000, st_size=4096, st_atime=1739794182, st_mtime=1739961403, st_ctime=1739961403)
INFO: mode: 0o40777

--- check: source_trees_without_genfiles
INFO: tensorboard_roots (1): ['[REDACTED_PATH]/.venv/lib/python3.10/site-packages']; bad_roots (0): []

--- check: full_pip_freeze
INFO: pip freeze --all:
absl-py==2.1.0
anyio==4.8.0
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
array_record==0.6.0
arrow==1.3.0
asttokens==3.0.0
astunparse==1.6.3
async-lru==2.0.4
attrs==25.1.0
babel==2.17.0
beautifulsoup4==4.13.3
bleach==6.2.0
certifi==2025.1.31
cffi==1.17.1
charset-normalizer==3.4.1
click==8.1.8
comm==0.2.2
debugpy==1.8.12
decorator==5.1.1
defusedxml==0.7.1
dm-tree==0.1.9
docstring_parser==0.16
etils==1.12.0
exceptiongroup==1.2.2
executing==2.2.0
fastjsonschema==2.21.1
flatbuffers==25.2.10
fqdn==1.5.1
fsspec==2025.2.0
gast==0.6.0
google-pasta==0.2.0
grpcio==1.70.0
gviz-api==1.10.0
h11==0.14.0
h5py==3.13.0
httpcore==1.0.7
httpx==0.28.1
idna==3.10
immutabledict==4.2.1
importlib_resources==6.5.2
ipykernel==6.29.5
ipython==8.32.0
ipywidgets==8.1.5
isoduration==20.11.0
jedi==0.19.2
Jinja2==3.1.5
json5==0.10.0
jsonpointer==3.0.0
jsonschema==4.23.0
jsonschema-specifications==2024.10.1
jupyter==1.1.1
jupyter-console==6.6.3
jupyter-events==0.12.0
jupyter-lsp==2.2.5
jupyter_client==8.6.3
jupyter_core==5.7.2
jupyter_server==2.15.0
jupyter_server_terminals==0.5.3
jupyterlab==4.3.5
jupyterlab_pygments==0.3.0
jupyterlab_server==2.27.3
jupyterlab_widgets==3.0.13
keras==3.8.0
libclang==18.1.1
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==3.0.2
matplotlib-inline==0.1.7
mdurl==0.1.2
mistune==3.1.2
ml-dtypes==0.4.1
namex==0.0.8
nbclient==0.10.2
nbconvert==7.16.6
nbformat==5.10.4
nest-asyncio==1.6.0
notebook==7.3.2
notebook_shim==0.2.4
numpy==2.0.2
nvidia-cublas-cu12==12.5.3.2
nvidia-cuda-cupti-cu12==12.5.82
nvidia-cuda-nvcc-cu12==12.5.82
nvidia-cuda-nvrtc-cu12==12.5.82
nvidia-cuda-runtime-cu12==12.5.82
nvidia-cudnn-cu12==9.3.0.75
nvidia-cufft-cu12==11.2.3.61
nvidia-curand-cu12==10.3.6.82
nvidia-cusolver-cu12==11.6.3.83
nvidia-cusparse-cu12==12.5.1.3
nvidia-nccl-cu12==2.21.5
nvidia-nvjitlink-cu12==12.5.82
opt_einsum==3.4.0
optree==0.14.0
overrides==7.7.0
packaging==24.2
pandocfilters==1.5.1
parso==0.8.4
pexpect==4.9.0
pip==22.0.2
platformdirs==4.3.6
prometheus_client==0.21.1
promise==2.3
prompt_toolkit==3.0.50
protobuf==3.20.3
psutil==7.0.0
ptyprocess==0.7.0
pure_eval==0.2.3
pyarrow==19.0.1
pycparser==2.22
Pygments==2.19.1
python-dateutil==2.9.0.post0
python-json-logger==3.2.1
PyYAML==6.0.2
pyzmq==26.2.1
referencing==0.36.2
requests==2.32.3
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==13.9.4
rpds-py==0.22.3
Send2Trash==1.8.3
setuptools==59.6.0
simple-parsing==0.1.7
six==1.17.0
sniffio==1.3.1
soupsieve==2.6
stack-data==0.6.3
tensorboard==2.18.0
tensorboard-data-server==0.7.2
tensorboard-plugin-profile==2.19.0
tensorflow==2.18.0
tensorflow-datasets==4.9.7
tensorflow-io-gcs-filesystem==0.37.1
tensorflow-metadata==1.16.1
termcolor==2.5.0
terminado==0.18.1
tinycss2==1.4.0
toml==0.10.2
tomli==2.2.1
tornado==6.4.2
tqdm==4.67.1
traitlets==5.14.3
types-python-dateutil==2.9.0.20241206
typing_extensions==4.12.2
uri-template==1.3.0
urllib3==2.3.0
wcwidth==0.2.13
webcolors==24.11.1
webencodings==0.5.1
websocket-client==1.8.0
Werkzeug==3.1.3
wheel==0.45.1
widgetsnbextension==4.0.13
wrapt==1.17.2
zipp==3.21.0

``````

</details>

## Issue description

I tried to test the Tensorboard profiler. I followed the [tutorial](https://www.tensorflow.org/tensorboard/tensorboard_profiling_keras) and created the following script:
<details>
<summary>profiler_test.py</summary>

``````python
from datetime import datetime
from packaging import version
import os
import tensorflow as tf

print("TensorFlow version: ", tf.__version__)

device_name = tf.test.gpu_device_name()
if not device_name:
  raise SystemError('GPU device not found')
print('Found GPU at: {}'.format(device_name))

import tensorflow_datasets as tfds
tfds.disable_progress_bar()

(ds_train, ds_test), ds_info = tfds.load(
    'mnist',
    split=['train', 'test'],
    shuffle_files=True,
    as_supervised=True,
    with_info=True,
)

def normalize_img(image, label):
  """Normalizes images: `uint8` -> `float32`."""
  return tf.cast(image, tf.float32) / 255., label

ds_train = ds_train.map(normalize_img)
ds_train = ds_train.batch(128)

model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28, 1)),
  tf.keras.layers.Dense(128,activation='relu'),
  tf.keras.layers.Dense(10, activation='softmax')
])
model.compile(
    loss='sparse_categorical_crossentropy',
    optimizer=tf.keras.optimizers.Adam(0.001),
    metrics=['accuracy']
)

logs = "logs/" + datetime.now().strftime("%Y%m%d-%H%M%S")

tboard_callback = tf.keras.callbacks.TensorBoard(log_dir = logs,
                                                 histogram_freq = 1,
                                                 profile_batch = '500,520')

model.fit(ds_train,
          epochs=2,
          validation_data=ds_test,
          callbacks = [tboard_callback])
``````
</details>

Once the script finishes, I run ```tensorboard --logdir=logs --bind_all``` to start Tensorboard. Tensorboard then starts as expected and is reachable. Once I click on the Profiler tab in Tensorboard though, it crashes and I get the following console output:

<details>
<summary>Console output</summary>

``````
W0219 10:36:53.961225 140124482418240 security_validator.py:60] In 3.0, this warning will become an error:
Illegal Content-Security-Policy for script-src: 'unsafe-inline'
Illegal Content-Security-Policy for connect-src: data:
Illegal Content-Security-Policy for connect-src: www.gstatic.com
Illegal Content-Security-Policy for script-src-elem: 'unsafe-inline'
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1739961418.129995 4173284 xplane_to_tools_data.cc:376] serving tool: tool_names with options: {}
I0000 00:00:1739961418.503777 4173287 xplane_to_tools_data.cc:376] serving tool: overview_page with options: {}
W0000 00:00:1739961419.334407 4173287 xplane_to_op_stats.cc:364] No CoreDetails found for TPU device plane: /device:GPU:0
2025-02-19 10:36:59.486345: F external/org_tensorflow/tensorflow/core/profiler/utils/kernel_stats_utils.cc:339] Check failed: stats.is_op_tensor_core_eligible == kernel_report.is_op_tensor_core_eligible() (1 vs. 0)
Fatal Python error: Aborted

Current thread 0x00007f71467fc640 (most recent call first):
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard_plugin_profile/convert/raw_to_tool_data.py", line 132 in xspace_to_tool_data
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard_plugin_profile/profile_plugin.py", line 673 in data_impl
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard_plugin_profile/profile_plugin.py", line 704 in data_route
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/werkzeug/wrappers/request.py", line 193 in application
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/backend/application.py", line 551 in _route_request
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/backend/auth_context_middleware.py", line 38 in __call__
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/backend/client_feature_flags.py", line 55 in __call__
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/backend/empty_path_redirect.py", line 43 in __call__
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/backend/experiment_id.py", line 73 in __call__
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/backend/path_prefix.py", line 68 in __call__
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/backend/security_validator.py", line 91 in __call__
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/backend/application.py", line 569 in wrapper
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/backend/application.py", line 528 in __call__
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/werkzeug/serving.py", line 331 in execute
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/werkzeug/serving.py", line 370 in run_wsgi
  File "/usr/lib/python3.10/http/server.py", line 421 in handle_one_request
  File "/usr/lib/python3.10/http/server.py", line 433 in handle
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/werkzeug/serving.py", line 398 in handle
  File "/usr/lib/python3.10/socketserver.py", line 747 in __init__
  File "/usr/lib/python3.10/socketserver.py", line 360 in finish_request
  File "/usr/lib/python3.10/socketserver.py", line 683 in process_request_thread
  File "/usr/lib/python3.10/threading.py", line 953 in run
  File "/usr/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
  File "/usr/lib/python3.10/threading.py", line 973 in _bootstrap

Thread 0x00007f72cdafd000 (most recent call first):
  File "/usr/lib/python3.10/selectors.py", line 416 in select
  File "/usr/lib/python3.10/socketserver.py", line 232 in serve_forever
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/werkzeug/serving.py", line 819 in serve_forever
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/program.py", line 297 in _run_serve_subcommand
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/program.py", line 278 in main
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/absl/app.py", line 254 in _run_main
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/absl/app.py", line 308 in run
  File "[REDACTED_PATH]/.venv/lib/python3.10/site-packages/tensorboard/main.py", line 41 in run_main
  File "[REDACTED_PATH]/.venv/bin/tensorboard", line 8 in <module>

Extension modules: markupsafe._speedups, google.protobuf.pyext._message, numpy._core._multiarray_umath, numpy._core._multiarray_tests, numpy.linalg._umath_linalg, grpc._cython.cygrpc, charset_normalizer.md, requests.packages.charset_normalizer.md, requests.packages.chardet.md, h5py._errors, h5py.defs, h5py._objects, h5py.h5, h5py.utils, h5py.h5t, h5py.h5s, h5py.h5ac, h5py.h5p, h5py.h5r, h5py._proxy, h5py._conv, h5py.h5z, h5py.h5a, h5py.h5d, h5py.h5ds, h5py.h5g, h5py.h5i, h5py.h5o, h5py.h5f, h5py.h5fd, h5py.h5pl, h5py.h5l, h5py._selector, psutil._psutil_linux, psutil._psutil_posix (total: 35)
Aborted (core dumped)
``````
</details>

**Comment by chriscarollo** [comment]
I'm having the same error.  No idea how to deal with it.

**Comment by chriscarollo** [comment]
It seems like my issue was that I was installing tensorflow/tensorflow:12.6.1, but was not specifying a version when I was pip installing tensorboard_plugin_profile.  Adding a ==2.16.0 appears to have fixed my crash.

A little bit of version-compatibility checking would go a long way towards user-friendliness.

**Comment by j-irion** [comment]
Thanks a lot! That seems to do the trick!
